morph/cli.py

```diff
@@ -78,6 +80,23 @@ def db_path(db_name):
     return os.path.join(profile_path(), 'dbs', db_name + '.db')
 
 
+def load_db(db_name, allow_missing=False):
+    path = db_path(db_name)
+    if not allow_missing and not os.access(path, os.R_OK):
+        die('can\'t read db file: %s' % (path,))
+    return MorphDb(path, ignoreErrors=allow_missing)
+
+
+def parse_morpheme(morpheme_str):
+    fields = morpheme_str.split('\t')
+    if len(fields) != 4:
+        die('bad morpheme (should be 4 tab-separated fields, MorphMan-style): %s'
+            % (morpheme_str,))
+    # Only 4 fields show up in equality, hash, or show, but the constructor
+    # takes an extra one.  That's the second argument; duplicate the first.
```

Comment:
Very interesting! Yeah, I noticed that the code seemed a little confused about this, but hadn't quite worked out all the ways it is. Thanks for the nice explanation :)

In particular, I'd missed that there were parts of the code that actually use the `inflected` attribute! It looks like that's only in highlighting, fortunately. Using that attribute seems pretty much doomed to be buggy given this confusion.

I suspect one thing that's going on with the で / だ example is that the て-form of だ is で, and in many sentences where a morpheme spelled/pronounced で appears, it's as the て-form of だ. For example, in 静かで便利 "quiet and convenient":
```
$ echo 静かで便利 | mecab
静か	名詞,形容動詞語幹,*,*,*,*,静か,シズカ,シズカ
で	助動詞,*,*,*,特殊・ダ,連用形,だ,デ,デ
便利	名詞,形容動詞語幹,*,*,*,*,便利,ベンリ,ベンリ
EOS
```
where mecab identifies the で as a form of the 助動詞 (auxiliary verb) だ. By contrast, in 日本でいる "be in Japan":
```
$ echo 日本でいる | mecab
日本	名詞,固有名詞,地域,国,*,*,日本,ニッポン,ニッポン
で	助詞,格助詞,一般,*,*,*,で,デ,デ
いる	動詞,自立,*,*,五段・ラ行,基本形,いる,イル,イル
EOS
```
it's a particle, specifically a 格助詞 which my dictionary glosses as "case-marking particle", and the base form is just で.

I *think* this should be equally broken for any lexeme that appears in multiple inflections. Maybe it's especially bad for だ / で just because it's so ubiquitous, you're sure to see it many times in a session. If this diagnosis is right, I'd predict that if you recalculate with Ctrl-M at the start of a study session, you'll see this bug with lots of moderately-common words, because you'd have seen them there and memoized a random one of the inflections.

I am suspicious of the logic of `fixReading` in another way or two -- for example it assumes that the base, in isolation, will have the reading that corresponds to the actual morpheme that mecab identified in context in the given expression. I am not quite coming up right now with a demo of this getting the wrong answer, and it's 0126 in my timezone and I should sleep instead of pursuing that further right now :-p, but I am about 97% confident that that can be wrong.

> The cleanest way is to create a `Lexeme` class, which does not have the `inflected` attribute and remove the `__eq__` and `__hash__` functions from the `Morpheme`.
> 
> This probably means that this project should have been called LexMan instead of MorphMan :)

Agreed :)